src/social_wallet_api/handler.clj

```diff
@@ -293,15 +300,32 @@ Creates a transaction.
                                                         (:amount query)
                                                         (:to-id query)
                                                         (dissoc query :tags))]
-                           ;; store to db as well with transaction-id
-                           (lib/create-transaction (get-db-blockchain blockchains)
-                                                   (:from-id query)
-                                                   (:amount query)
-                                                   (:to-id query)
-                                                   (-> query 
-                                                       (dissoc :comment :comment-to)
-                                                       (assoc :transaction-id transaction-id)
-                                                       (assoc :currency (:blockchain query))))
+                           (do
+                             ;; Update fee to db when confirmed
+                             ;; The logged-future will return an exception which otherwise would be swallowed till deref
+                             (log/logged-future
+                              (while (> (:number-confirmations blockchain)
+                                        (number-confirmations blockchain transaction-id))
+                                 (log/debug "Not enough confirmations for transaction with id " transaction-id)
+                                 (Thread/sleep 20000))
```

Comment:
let's make also this number a configuration directive like `frequency-confirmations`

Reply:
will do